### ISSUE TITLE
getDeclaredVariables does not return switch discriminant

### ISSUE THREAD

**Issue by alecxe** [eslint/eslint#6508]
The `context.getDeclaredVariables()` function does not return the `node.discriminant` in case of a `SwitchStatement`.

Here is the rule I'm working on:

```javascript
module.exports = function (context) {
  var protractorGlobals = [
    'browser',
    'protractor',
    'element',
    'by',
    '$',
    '$$'
  ]

  function checkVariables (node) {
    var variables = context.getDeclaredVariables(node)  
    console.log(variables)  // this would be empty in case of SwitchStatement
    for (var i = 0; i < variables.length; ++i) {
      if (protractorGlobals.indexOf(variables[i].name) !== -1) {
        context.report(node, 'Unexpected Protractor built-in global variable shadowing')
      }
    }
  }

  return {
    'VariableDeclaration': checkVariables,
    'FunctionDeclaration': checkVariables,
    'FunctionExpression': checkVariables,
    'CatchClause': checkVariables,
    'SwitchStatement': checkVariables
  }
}
```

If you execute the rule against `switch (element) { case 1: break; default: break; }` code, `context.getDeclaredVariables(node) ` would return an empty array.

Is this a bug or just my misunderstanding? Thanks.

**Comment by mysticatea** [comment]
Yes, it's expected behavior. SwitchStatement never defines any variables. That `element` is a reference, not a variable declaration.

Details: https://github.com/eslint/eslint/blob/64b0d0c26cf1c517648253e051f827c09868a214/lib/eslint.js#L1167

**Comment by alecxe** [comment]
@mysticatea okay, gotcha.

Do you think this was a good way to handle that: https://github.com/alecxe/eslint-plugin-protractor/blob/master/lib/rules/no-shadowing.js#L26?

Thanks.

**Comment by mysticatea** [comment]
No. It's not shadowing. It's a reference.

**Comment by alecxe** [comment]
Ah, you are right. I should not be handling `switch` in the first place. In all other cases which the rule handles, I think this would be shadowing. Thanks!
